pkg/services/authn/grpcutils/grpc_authenticator.go

```diff
@@ -117,36 +117,42 @@ func (f *AuthenticatorWithFallback) Authenticate(ctx context.Context) (context.C
 	if err != nil {
 		// In case of error, fallback to the legacy authenticator
 		newCtx, err = f.fallback.Authenticate(ctx)
-		f.metrics.fallbackCounter.WithLabelValues(fmt.Sprintf("%t", err == nil)).Inc()
 		span.SetAttributes(attribute.Bool("fallback_used", true))
+		if err != nil {
+			f.metrics.requestsTotal.WithLabelValues("true", "false").Inc()
+			return nil, err
+		}
+		f.metrics.requestsTotal.WithLabelValues("true", fmt.Sprintf("%t", err == nil)).Inc()
```

Comment:
nit: here and below feels like there's no need to compare the error, as we know that it's nil. So could just do:
```suggestion
		f.metrics.requestsTotal.WithLabelValues("true", "true").Inc()
```